src/Config.hs

```diff
@@ -261,21 +273,28 @@ mkPackage (CaptureUnknownFields unknownFields PackageConfig{..}) = do
       where
         f name = "Specified source-dir " ++ show name ++ " does not exist"
 
-    github = ("https://github.com/" ++) <$> packageConfigGithub
+    github :: Maybe GithubConfig
+    github = modifyGithubUrl ("https://github.com/" <>) <$>
+             maybe Nothing (Just . splitGithub) packageConfigGithub
+      where
+        splitGithub str = case T.unpack <$> T.split (== '/') str of
+          [user, repo, subdir] ->
+            GithubConfig (user <> "/" <> repo) (Just subdir)
+          _ -> GithubConfig (T.unpack str) Nothing
 
     homepage :: Maybe String
     homepage = case packageConfigHomepage of
       Just Nothing -> Nothing
       _ -> join packageConfigHomepage <|> fromGithub
       where
-        fromGithub = ((++ "#readme") <$> github)
+        fromGithub = (<> "#readme") . githubConfigUrl <$> github
```

Comment:
As we stick with `String` I would prefer to not change this to `<>` (as `<>` is not available in older versions of base).